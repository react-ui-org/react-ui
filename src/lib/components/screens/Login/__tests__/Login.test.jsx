import React from 'react';
import { mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';
import { Login } from '..';

describe('rendering', () => {
  it('renders correctly', () => {
    const tree = mount(<Login />);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('renders correctly with all props except translations', () => {
    const tree = mount(<Login
      id="custom-id"
      logo="http://satyr.io/100x100/33?text=logo"
      title="Company"
      hasError
      footer={(
        <a href="http://example.com">link</a>
      )}
    />);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('renders correctly with username', () => {
    const tree = mount(<Login usernameType="Username" />);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('renders correctly with translations', () => {
    const tree = mount(<Login
      translations={{
        email: 'E-mail',
        invalidUsernameOrPassword: 'Nesprávný e-mail nebo heslo',
        password: 'Heslo',
        signIn: 'Přihlásit se',
      }}
    />);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});

describe('functionality', () => {
  it('calls submitHandler() on submit', () => {
    const spy = sinon.spy();
    const component = mount(<Login submitHandler={spy} />);

    component
      .find('form')
      .simulate('submit');
    expect(spy.calledOnce).toEqual(true);
  });

  it('calls onChangeHandler() on field change', () => {
    const spy = sinon.spy();
    const component = mount(<Login onChangeHandler={spy} />);

    component
      .find('input').first()
      .simulate('change');
    expect(spy.calledOnce).toEqual(true);
  });
});

