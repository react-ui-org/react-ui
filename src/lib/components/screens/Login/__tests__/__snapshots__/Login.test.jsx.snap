// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly 1`] = `
<WithTranslationContextComponent
  translations={null}
>
  <Login
    footer={null}
    hasError={false}
    logoUrl={null}
    onChangeHandler={null}
    submitHandler={null}
    title={null}
    translations={
      Object {
        "email": "E-mail",
        "invalidUsernameOrPassword": "Invalid username or password",
        "password": "Password",
        "signIn": "Sign in",
      }
    }
    usernameType="email"
  >
    <div
      className="root"
    >
      <div
        className="box"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="mb-3"
          >
            <ForwardRef(withForwardedRef(TextField))
              autoComplete="username"
              changeHandler={[Function]}
              fullWidth={true}
              id="usernameInput"
              label="E-mail"
              required={true}
              type="email"
            >
              <TextField
                autoComplete="username"
                changeHandler={[Function]}
                disabled={false}
                forwardedRef={null}
                fullWidth={true}
                helperText={null}
                id="usernameInput"
                inFormLayout={false}
                inputSize={null}
                isLabelVisible={true}
                label="E-mail"
                layout="vertical"
                placeholder={null}
                required={true}
                size="medium"
                type="email"
                validationState={null}
                variant="outline"
              >
                <label
                  className="root
        isRootFullWidth
        
        rootLayoutVertical
        isRootRequired
        rootSizeMedium
        
        
        rootVariantOutline"
                  htmlFor="usernameInput"
                  id="usernameInput__label"
                  style={Object {}}
                >
                  <div
                    className="label"
                    id="usernameInput__labelText"
                  >
                    E-mail
                  </div>
                  <div
                    className="field"
                  >
                    <div
                      className="inputContainer"
                    >
                      <input
                        autoComplete="username"
                        className="input"
                        disabled={false}
                        id="usernameInput"
                        onChange={[Function]}
                        placeholder={null}
                        required={true}
                        size={null}
                        type="email"
                      />
                    </div>
                  </div>
                </label>
              </TextField>
            </ForwardRef(withForwardedRef(TextField))>
            <ForwardRef(withForwardedRef(TextField))
              autoComplete="current-password"
              changeHandler={[Function]}
              fullWidth={true}
              id="passwordInput"
              label="Password"
              required={true}
              type="password"
            >
              <TextField
                autoComplete="current-password"
                changeHandler={[Function]}
                disabled={false}
                forwardedRef={null}
                fullWidth={true}
                helperText={null}
                id="passwordInput"
                inFormLayout={false}
                inputSize={null}
                isLabelVisible={true}
                label="Password"
                layout="vertical"
                placeholder={null}
                required={true}
                size="medium"
                type="password"
                validationState={null}
                variant="outline"
              >
                <label
                  className="root
        isRootFullWidth
        
        rootLayoutVertical
        isRootRequired
        rootSizeMedium
        
        
        rootVariantOutline"
                  htmlFor="passwordInput"
                  id="passwordInput__label"
                  style={Object {}}
                >
                  <div
                    className="label"
                    id="passwordInput__labelText"
                  >
                    Password
                  </div>
                  <div
                    className="field"
                  >
                    <div
                      className="inputContainer"
                    >
                      <input
                        autoComplete="current-password"
                        className="input"
                        disabled={false}
                        id="passwordInput"
                        onChange={[Function]}
                        placeholder={null}
                        required={true}
                        size={null}
                        type="password"
                      />
                    </div>
                  </div>
                </label>
              </TextField>
            </ForwardRef(withForwardedRef(TextField))>
          </div>
          <ForwardRef(withForwardedRef(Button))
            block={true}
            label="Sign in"
            type="submit"
          >
            <Button
              afterLabel={null}
              beforeLabel={null}
              block={true}
              clickHandler={null}
              disabled={false}
              endCorner={null}
              forwardedRef={null}
              grouped={false}
              label="Sign in"
              labelVisibility="all"
              loadingIcon={null}
              priority="default"
              size="medium"
              startCorner={null}
              type="submit"
              variant="primary"
            >
              <button
                className="root
        priorityDefault
        sizeMedium
        variantPrimary
        isRootBlock"
                disabled={false}
                onClick={null}
                type="submit"
              >
                <span
                  className="label"
                >
                  Sign in
                </span>
              </button>
            </Button>
          </ForwardRef(withForwardedRef(Button))>
        </form>
      </div>
    </div>
  </Login>
</WithTranslationContextComponent>
`;

exports[`rendering renders correctly with all props except translations 1`] = `
<WithTranslationContextComponent
  footer={
    <a
      href="http://example.com"
    >
      link
    </a>
  }
  hasError={true}
  id="custom-id"
  logo="http://satyr.io/100x100/33?text=logo"
  title="Company"
  translations={null}
>
  <Login
    footer={
      <a
        href="http://example.com"
      >
        link
      </a>
    }
    hasError={true}
    id="custom-id"
    logo="http://satyr.io/100x100/33?text=logo"
    logoUrl={null}
    onChangeHandler={null}
    submitHandler={null}
    title="Company"
    translations={
      Object {
        "email": "E-mail",
        "invalidUsernameOrPassword": "Invalid username or password",
        "password": "Password",
        "signIn": "Sign in",
      }
    }
    usernameType="email"
  >
    <div
      className="root"
      id="custom-id"
    >
      <div
        className="title"
        id="custom-id__title"
      >
        Company
      </div>
      <div
        className="box"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="error"
            id="custom-id__errorText"
          >
            Invalid username or password
          </div>
          <div
            className="mb-3"
          >
            <ForwardRef(withForwardedRef(TextField))
              autoComplete="username"
              changeHandler={[Function]}
              fullWidth={true}
              id="custom-id__usernameInput"
              label="E-mail"
              required={true}
              type="email"
            >
              <TextField
                autoComplete="username"
                changeHandler={[Function]}
                disabled={false}
                forwardedRef={null}
                fullWidth={true}
                helperText={null}
                id="custom-id__usernameInput"
                inFormLayout={false}
                inputSize={null}
                isLabelVisible={true}
                label="E-mail"
                layout="vertical"
                placeholder={null}
                required={true}
                size="medium"
                type="email"
                validationState={null}
                variant="outline"
              >
                <label
                  className="root
        isRootFullWidth
        
        rootLayoutVertical
        isRootRequired
        rootSizeMedium
        
        
        rootVariantOutline"
                  htmlFor="custom-id__usernameInput"
                  id="custom-id__usernameInput__label"
                  style={Object {}}
                >
                  <div
                    className="label"
                    id="custom-id__usernameInput__labelText"
                  >
                    E-mail
                  </div>
                  <div
                    className="field"
                  >
                    <div
                      className="inputContainer"
                    >
                      <input
                        autoComplete="username"
                        className="input"
                        disabled={false}
                        id="custom-id__usernameInput"
                        onChange={[Function]}
                        placeholder={null}
                        required={true}
                        size={null}
                        type="email"
                      />
                    </div>
                  </div>
                </label>
              </TextField>
            </ForwardRef(withForwardedRef(TextField))>
            <ForwardRef(withForwardedRef(TextField))
              autoComplete="current-password"
              changeHandler={[Function]}
              fullWidth={true}
              id="custom-id__passwordInput"
              label="Password"
              required={true}
              type="password"
            >
              <TextField
                autoComplete="current-password"
                changeHandler={[Function]}
                disabled={false}
                forwardedRef={null}
                fullWidth={true}
                helperText={null}
                id="custom-id__passwordInput"
                inFormLayout={false}
                inputSize={null}
                isLabelVisible={true}
                label="Password"
                layout="vertical"
                placeholder={null}
                required={true}
                size="medium"
                type="password"
                validationState={null}
                variant="outline"
              >
                <label
                  className="root
        isRootFullWidth
        
        rootLayoutVertical
        isRootRequired
        rootSizeMedium
        
        
        rootVariantOutline"
                  htmlFor="custom-id__passwordInput"
                  id="custom-id__passwordInput__label"
                  style={Object {}}
                >
                  <div
                    className="label"
                    id="custom-id__passwordInput__labelText"
                  >
                    Password
                  </div>
                  <div
                    className="field"
                  >
                    <div
                      className="inputContainer"
                    >
                      <input
                        autoComplete="current-password"
                        className="input"
                        disabled={false}
                        id="custom-id__passwordInput"
                        onChange={[Function]}
                        placeholder={null}
                        required={true}
                        size={null}
                        type="password"
                      />
                    </div>
                  </div>
                </label>
              </TextField>
            </ForwardRef(withForwardedRef(TextField))>
          </div>
          <ForwardRef(withForwardedRef(Button))
            block={true}
            id="custom-id__signInButton"
            label="Sign in"
            type="submit"
          >
            <Button
              afterLabel={null}
              beforeLabel={null}
              block={true}
              clickHandler={null}
              disabled={false}
              endCorner={null}
              forwardedRef={null}
              grouped={false}
              id="custom-id__signInButton"
              label="Sign in"
              labelVisibility="all"
              loadingIcon={null}
              priority="default"
              size="medium"
              startCorner={null}
              type="submit"
              variant="primary"
            >
              <button
                className="root
        priorityDefault
        sizeMedium
        variantPrimary
        isRootBlock"
                disabled={false}
                id="custom-id__signInButton"
                onClick={null}
                type="submit"
              >
                <span
                  className="label"
                  id="custom-id__signInButton__labelText"
                >
                  Sign in
                </span>
              </button>
            </Button>
          </ForwardRef(withForwardedRef(Button))>
        </form>
        <div
          className="footer"
          id="custom-id__footerContent"
        >
          <a
            href="http://example.com"
          >
            link
          </a>
        </div>
      </div>
    </div>
  </Login>
</WithTranslationContextComponent>
`;

exports[`rendering renders correctly with translations 1`] = `
<WithTranslationContextComponent
  translations={
    Object {
      "email": "E-mail",
      "invalidUsernameOrPassword": "Nesprávný e-mail nebo heslo",
      "password": "Heslo",
      "signIn": "Přihlásit se",
    }
  }
>
  <Login
    footer={null}
    hasError={false}
    logoUrl={null}
    onChangeHandler={null}
    submitHandler={null}
    title={null}
    translations={
      Object {
        "email": "E-mail",
        "invalidUsernameOrPassword": "Nesprávný e-mail nebo heslo",
        "password": "Heslo",
        "signIn": "Přihlásit se",
      }
    }
    usernameType="email"
  >
    <div
      className="root"
    >
      <div
        className="box"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="mb-3"
          >
            <ForwardRef(withForwardedRef(TextField))
              autoComplete="username"
              changeHandler={[Function]}
              fullWidth={true}
              id="usernameInput"
              label="E-mail"
              required={true}
              type="email"
            >
              <TextField
                autoComplete="username"
                changeHandler={[Function]}
                disabled={false}
                forwardedRef={null}
                fullWidth={true}
                helperText={null}
                id="usernameInput"
                inFormLayout={false}
                inputSize={null}
                isLabelVisible={true}
                label="E-mail"
                layout="vertical"
                placeholder={null}
                required={true}
                size="medium"
                type="email"
                validationState={null}
                variant="outline"
              >
                <label
                  className="root
        isRootFullWidth
        
        rootLayoutVertical
        isRootRequired
        rootSizeMedium
        
        
        rootVariantOutline"
                  htmlFor="usernameInput"
                  id="usernameInput__label"
                  style={Object {}}
                >
                  <div
                    className="label"
                    id="usernameInput__labelText"
                  >
                    E-mail
                  </div>
                  <div
                    className="field"
                  >
                    <div
                      className="inputContainer"
                    >
                      <input
                        autoComplete="username"
                        className="input"
                        disabled={false}
                        id="usernameInput"
                        onChange={[Function]}
                        placeholder={null}
                        required={true}
                        size={null}
                        type="email"
                      />
                    </div>
                  </div>
                </label>
              </TextField>
            </ForwardRef(withForwardedRef(TextField))>
            <ForwardRef(withForwardedRef(TextField))
              autoComplete="current-password"
              changeHandler={[Function]}
              fullWidth={true}
              id="passwordInput"
              label="Heslo"
              required={true}
              type="password"
            >
              <TextField
                autoComplete="current-password"
                changeHandler={[Function]}
                disabled={false}
                forwardedRef={null}
                fullWidth={true}
                helperText={null}
                id="passwordInput"
                inFormLayout={false}
                inputSize={null}
                isLabelVisible={true}
                label="Heslo"
                layout="vertical"
                placeholder={null}
                required={true}
                size="medium"
                type="password"
                validationState={null}
                variant="outline"
              >
                <label
                  className="root
        isRootFullWidth
        
        rootLayoutVertical
        isRootRequired
        rootSizeMedium
        
        
        rootVariantOutline"
                  htmlFor="passwordInput"
                  id="passwordInput__label"
                  style={Object {}}
                >
                  <div
                    className="label"
                    id="passwordInput__labelText"
                  >
                    Heslo
                  </div>
                  <div
                    className="field"
                  >
                    <div
                      className="inputContainer"
                    >
                      <input
                        autoComplete="current-password"
                        className="input"
                        disabled={false}
                        id="passwordInput"
                        onChange={[Function]}
                        placeholder={null}
                        required={true}
                        size={null}
                        type="password"
                      />
                    </div>
                  </div>
                </label>
              </TextField>
            </ForwardRef(withForwardedRef(TextField))>
          </div>
          <ForwardRef(withForwardedRef(Button))
            block={true}
            label="Přihlásit se"
            type="submit"
          >
            <Button
              afterLabel={null}
              beforeLabel={null}
              block={true}
              clickHandler={null}
              disabled={false}
              endCorner={null}
              forwardedRef={null}
              grouped={false}
              label="Přihlásit se"
              labelVisibility="all"
              loadingIcon={null}
              priority="default"
              size="medium"
              startCorner={null}
              type="submit"
              variant="primary"
            >
              <button
                className="root
        priorityDefault
        sizeMedium
        variantPrimary
        isRootBlock"
                disabled={false}
                onClick={null}
                type="submit"
              >
                <span
                  className="label"
                >
                  Přihlásit se
                </span>
              </button>
            </Button>
          </ForwardRef(withForwardedRef(Button))>
        </form>
      </div>
    </div>
  </Login>
</WithTranslationContextComponent>
`;

exports[`rendering renders correctly with username 1`] = `
<WithTranslationContextComponent
  translations={null}
  usernameType="Username"
>
  <Login
    footer={null}
    hasError={false}
    logoUrl={null}
    onChangeHandler={null}
    submitHandler={null}
    title={null}
    translations={
      Object {
        "email": "E-mail",
        "invalidUsernameOrPassword": "Invalid username or password",
        "password": "Password",
        "signIn": "Sign in",
      }
    }
    usernameType="Username"
  >
    <div
      className="root"
    >
      <div
        className="box"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="mb-3"
          >
            <ForwardRef(withForwardedRef(TextField))
              autoComplete="username"
              changeHandler={[Function]}
              fullWidth={true}
              id="usernameInput"
              label="Username"
              required={true}
              type="text"
            >
              <TextField
                autoComplete="username"
                changeHandler={[Function]}
                disabled={false}
                forwardedRef={null}
                fullWidth={true}
                helperText={null}
                id="usernameInput"
                inFormLayout={false}
                inputSize={null}
                isLabelVisible={true}
                label="Username"
                layout="vertical"
                placeholder={null}
                required={true}
                size="medium"
                type="text"
                validationState={null}
                variant="outline"
              >
                <label
                  className="root
        isRootFullWidth
        
        rootLayoutVertical
        isRootRequired
        rootSizeMedium
        
        
        rootVariantOutline"
                  htmlFor="usernameInput"
                  id="usernameInput__label"
                  style={Object {}}
                >
                  <div
                    className="label"
                    id="usernameInput__labelText"
                  >
                    Username
                  </div>
                  <div
                    className="field"
                  >
                    <div
                      className="inputContainer"
                    >
                      <input
                        autoComplete="username"
                        className="input"
                        disabled={false}
                        id="usernameInput"
                        onChange={[Function]}
                        placeholder={null}
                        required={true}
                        size={null}
                        type="text"
                      />
                    </div>
                  </div>
                </label>
              </TextField>
            </ForwardRef(withForwardedRef(TextField))>
            <ForwardRef(withForwardedRef(TextField))
              autoComplete="current-password"
              changeHandler={[Function]}
              fullWidth={true}
              id="passwordInput"
              label="Password"
              required={true}
              type="password"
            >
              <TextField
                autoComplete="current-password"
                changeHandler={[Function]}
                disabled={false}
                forwardedRef={null}
                fullWidth={true}
                helperText={null}
                id="passwordInput"
                inFormLayout={false}
                inputSize={null}
                isLabelVisible={true}
                label="Password"
                layout="vertical"
                placeholder={null}
                required={true}
                size="medium"
                type="password"
                validationState={null}
                variant="outline"
              >
                <label
                  className="root
        isRootFullWidth
        
        rootLayoutVertical
        isRootRequired
        rootSizeMedium
        
        
        rootVariantOutline"
                  htmlFor="passwordInput"
                  id="passwordInput__label"
                  style={Object {}}
                >
                  <div
                    className="label"
                    id="passwordInput__labelText"
                  >
                    Password
                  </div>
                  <div
                    className="field"
                  >
                    <div
                      className="inputContainer"
                    >
                      <input
                        autoComplete="current-password"
                        className="input"
                        disabled={false}
                        id="passwordInput"
                        onChange={[Function]}
                        placeholder={null}
                        required={true}
                        size={null}
                        type="password"
                      />
                    </div>
                  </div>
                </label>
              </TextField>
            </ForwardRef(withForwardedRef(TextField))>
          </div>
          <ForwardRef(withForwardedRef(Button))
            block={true}
            label="Sign in"
            type="submit"
          >
            <Button
              afterLabel={null}
              beforeLabel={null}
              block={true}
              clickHandler={null}
              disabled={false}
              endCorner={null}
              forwardedRef={null}
              grouped={false}
              label="Sign in"
              labelVisibility="all"
              loadingIcon={null}
              priority="default"
              size="medium"
              startCorner={null}
              type="submit"
              variant="primary"
            >
              <button
                className="root
        priorityDefault
        sizeMedium
        variantPrimary
        isRootBlock"
                disabled={false}
                onClick={null}
                type="submit"
              >
                <span
                  className="label"
                >
                  Sign in
                </span>
              </button>
            </Button>
          </ForwardRef(withForwardedRef(Button))>
        </form>
      </div>
    </div>
  </Login>
</WithTranslationContextComponent>
`;
